libp2p/protocols/pubsub/floodsub.nim

```diff
@@ -106,10 +106,9 @@ method unsubscribePeer*(f: FloodSub, peer: PeerId) =
 method rpcHandler*(f: FloodSub,
                    peer: PubSubPeer,
                    data: seq[byte]) {.async.} =
-
   var rpcMsg = decodeRpcMsg(data).valueOr:
     debug "failed to decode msg from peer", peer, err = error
-    raise newException(CatchableError, "")
+    raise newException(CatchableError, "Peer msg couldn't be decoded")
```

Comment:
I believe we should either log and discard the exception or create a sequence Parent/Child that gets logged only once. Unfortunately, this seems not possible in Nim. I created this issue some time ago, but it was closed without a solution https://github.com/nim-lang/Nim/issues/20677. @arnetheduck thought?

Reply:
you can: `(ref CatchableError)(mgs: "test", parent: otherException)` - when catching, you can use the parent field to print all parents as well - this latter thing does not exist but the core support in nim is there.